client/my-sites/stats/geochart/index.jsx

```diff
@@ -328,10 +390,14 @@ class StatsGeochart extends Component {
 			domain: currentUserCountryCode,
 		};
 
-		if ( geoMode !== 'country' ) {
+		if ( geoMode === 'city' ) {
```

Comment:
I'm wondering if these conditional and other related locations decisions could be better represented in a single switch/case, making it clear what applies to `country`, `regions` or `city` locations. My impression is that this would be considerably more straightforward than using different `if` statements on multiple parts of the code, and also would not require to swap the logic for similar refactorings in the future.

Still, that would represent a significant code reorganization and can be done in a separate PR, if we decide for that.

Reply:
Thanks for sharing this! I think this is something we could iterate on once we have the region coordinates. I’ll add it to my to-do list to track it for later.